internal/docs/design.md

```diff
@@ -105,14 +105,17 @@ https://godoc.org/github.com/google/go-cloud/runtimevar#Variable
 https://godoc.org/github.com/google/go-cloud/blob#Bucket.NewWriter
 [`database/sql`]: https://godoc.org/database/sql
 
-## No Global State
+## Minimize Global State
 
 As a library, the Go CDK should not introduce global state. Global state is
 difficult to reason about in large codebases, where it can be necessary for
 different parts of the application to use different states. Instead of adding
 global state, push responsibility to the application to inject the state where
 it is needed.
 
+The exception we permit is URL scheme registration as documented under
```

Comment:
Why do we permit it?

Reply:
@vangent, I'm having trouble coming up with a crisp rationale here, but I think @ijt's sentiment indicates we should have one written here. I'll try to talk around it and see if we stumble upon the right phrasing:

We found that the setup code for a non-Wire user wanting to use multiple providers with URLs was pretty painful. The concession here is that we permit registration for the 90% use-case, but any level of customization requires creating the registry from scratch. URL-opening is built on top of other constructors, so using this logic does not require global state.

Reply:
I guess the rationale is that it just makes URLs so easy to use. There's no way to achieve that without letting the package initializers set things up, which requires global state. And, we're providing an alternative for more advanced users who want to avoid it.